File: shoring/flywheel/src/test/java/com/panopset/tests/flywheel/BufferTest.java
package com.panopset.tests.flywheel;

import java.io.IOException;

import org.junit.jupiter.api.Test;

import com.panopset.compat.Stringop;
import com.panopset.tests.transformer.FlywheelTemplateToFileTest;
import com.panopset.tests.transformer.FlywheelTemplateToTextTransformerTest;
import com.panopset.tests.transformer.StandardPackagePath;

public final class BufferTest {

	public static final String SIMPLEONECHAR = "simpleOneChar.txt";
	public static final String SIMPLETWOLINES = "simpleTwoLines.txt";

	@Test
	void testSimpleOneChar() throws IOException {
		String expected = String.format("x%s", Stringop.getEol());
		new FlywheelTemplateToTextTransformerTest(this.getClass().getPackageName(), SIMPLEONECHAR, expected).test();
	}

	@Test
	void testTwoLines() throws IOException {
		String expected = "x" + Stringop.getEol() + "y" + Stringop.getEol();
		new FlywheelTemplateToTextTransformerTest(this.getClass().getPackageName(), SIMPLETWOLINES, expected).test();
	}

	@Test
	void testSimpleBuffer() throws IOException {
		new FlywheelTemplateToFileTest(this.getClass().getPackageName(), SimpleTest.SIMPLETEST, SimpleTest.SIMPLEOUT, SimpleTest.EXPECTED).test();
		String expected = new StandardPackagePath(this.getClass().getPackageName()).getRezStr(SimpleTest.EXPECTED);
		new FlywheelTemplateToTextTransformerTest(this.getClass().getPackageName(), SimpleTest.SIMPLETEST, expected).test();
		new FlywheelTemplateToFileTest(this.getClass().getPackageName(), SimpleTest.SIMPLETEST, SimpleTest.SIMPLEOUT, SimpleTest.EXPECTED).test();
	}

	@Test
	void testComplexBuffer() throws IOException {
		String expected = new StandardPackagePath(this.getClass().getPackageName()).getRezStr(ComplexTest.EXPECTED);
		new FlywheelTemplateToTextTransformerTest(this.getClass().getPackageName(), ComplexTest.TEMPLATE, expected).test();
		new FlywheelTemplateToFileTest(this.getClass().getPackageName(), ComplexTest.TEMPLATE, ComplexTest.OUTPUT, ComplexTest.EXPECTED).test();
	}
}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/flywheel/BufferTest.kt
package com.panopset.tests.flywheel

import com.panopset.compat.Stringop.getEol
import com.panopset.tests.transformer.FlywheelTemplateToFileTest
import com.panopset.tests.transformer.FlywheelTemplateToTextTransformerTest
import com.panopset.tests.transformer.StandardPackagePath
import org.junit.jupiter.api.Test

class BufferTest {
    @Test
    fun testSimpleOneChar() {
        val expected = String.format("x")
        FlywheelTemplateToTextTransformerTest(this.javaClass.packageName, SIMPLEONECHAR, expected).test()
    }

    @Test
    fun testTwoLines() {
        val expected = "x" + getEol() + "y"
        FlywheelTemplateToTextTransformerTest(this.javaClass.packageName, SIMPLETWOLINES, expected).test()
    }

    @Test
    fun testSimpleBuffer() {
        FlywheelTemplateToFileTest(
            this.javaClass.packageName,
            SimpleTest.SIMPLETEST,
            SimpleTest.SIMPLEOUT,
            SimpleTest.EXPECTED
        ).test()
        val expected = StandardPackagePath(this.javaClass.packageName).getRezStr(SimpleTest.EXPECTED)
        FlywheelTemplateToTextTransformerTest(this.javaClass.packageName, SimpleTest.SIMPLETEST, expected).test()
        FlywheelTemplateToFileTest(
            this.javaClass.packageName,
            SimpleTest.SIMPLETEST,
            SimpleTest.SIMPLEOUT,
            SimpleTest.EXPECTED
        ).test()
    }

    @Test
    fun testComplexBuffer() {
        val expected = StandardPackagePath(this.javaClass.packageName).getRezStr(ComplexTest.EXPECTED)
        FlywheelTemplateToTextTransformerTest(this.javaClass.packageName, ComplexTest.TEMPLATE, expected).test()
        FlywheelTemplateToFileTest(
            this.javaClass.packageName,
            ComplexTest.TEMPLATE,
            ComplexTest.OUTPUT,
            ComplexTest.EXPECTED
        ).test()
    }

    companion object {
        const val SIMPLEONECHAR = "simpleOneChar.txt"
        const val SIMPLETWOLINES = "simpleTwoLines.txt"
    }
}
